### Q1. 항암약 위약을 투여 받은 환자의 부작용은 감기약 위약을 투여 받은 환자의 부작용 분포와 차이가 있는가?

항암약(위약)을 투여 받은 환자 그룹의 관찰된 부작용

1.   항목 추가
2.   항목 추가



In [ ]:
import pandas as pd
df = pd.DataFrame({ "항암약":[4,4,3,4,1,4,1,4,1,4,4,2,1,4,2,3,2,4,4,4] })
# 1: '아픔', 2: '조금 아픔', 3: '속쓰림', 4: '무증상'

감기약(위약)을 투여받은 환자들 중의 알려진 부작용 발생 비율


In [ ]:
# 1. 아픔: 10%
# 2. 조금 아픔 5%
# 3. 속 쓰림 15%
# 4. 무증상: 70%

-	귀무가설(H0): 감기약(위약)의 부작용과 항암약(위약)의 부작용은 동일하다.
- 	대립가설(H1): 감기약(위약)의 부작용과 항암약(위약)의 부작용은 다르다.

 ### 1-1. 항암약을 투여 받은 환자 중 '무증상'의 비율을 0과 1사이로 구하시오.

In [ ]:
#1번풀이
cond = df['항암약'] == 4
sum(cond) / len(df['항암약'])

0.55

In [ ]:
#2번풀이 벨류 카운드에 노말라이 제이션쓰면전체중에 비율을 나타내라는 것
df['항암약'].value_counts(normalize=True)

항암약
4    0.55
1    0.20
2    0.15
3    0.10
Name: proportion, dtype: float64

### 1-2. 감기약의 예상 부작용 비율과 항암약의 부작용 관찰값이 통계적으로 유의미하게 차이가 있는지 확인하려 한다. 카이 제곱 검정을 사용하여 검정 통계량을 구하시오.

이 경우는 관찰된 부작용 빈도(항암약)와 기대 빈도(감기약의 예상 부작용 비율에서 환자 수 기반으로 환산) 를 비교하는 것이므로, 일표본 카이제곱 검정(χ² 적합도 검정, goodness-of-fit test)에 해당합니다. 즉, 하나의 분포(예상 비율)에 대해 실제 관찰값이 얼마나 잘 들어맞는지 검정하는 방법입니다.

In [ ]:
ob = df['항암약'].value_counts().sort_index().to_list() #이건관찰빈도(observatory)   #인덱스 순으로 하고 , 카이제곱 검정을 하기위해 list형태로 만들어야함!!!   = 이게 오른쪽거랑 같음 = ob=[4,3,2,11]
ex = [0.1 * 20, 0.05 * 20, 0.15 * 20, 0.7 * 20] #이건 기대빈도수(expectaion)   #감기약부작용 발생비율(내가 발생한 실제 비율이 아니고, 우리가 정한 걸로하는것) * 관찰값의전체수(20)

from scipy import stats
stats.chisquare(ob, ex)


#ex의 0.1이런건 우리가 구하려는 기대 비



Power_divergenceResult(statistic=6.976190476190476, pvalue=0.07266054733847573)

In [ ]:
6.976190476190476

6.976190476190476

### 1-3. 위의 p-값을 구하시오

In [ ]:
0.07266054733847573

0.07266054733847573

### Q2. 다중 선형 회귀 문제

In [ ]:
#다중 선형회귀는 종속변수가 여러가지 독립변수에 얼마나 영향을 받는지 보는것임
# 그래서 밑에 문제는 종속변수인 온도가 태양에너지, 바람세기, 오존농도의 1단위 변화에 따라 어떻게 변하는지를 보는것임!
# 물류로 보면 지급율이, 배차횟수, 장거리근거리여부, 운송시간 등에 따라 지급율이 어떻게 변하는지 상관관계를 확인할 수 있음

In [ ]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/lovedlim/inf/refs/heads/main/p4/6_3/data6-3-2.csv")
df.head(2)

solar  wind     o3  temperature
0   89.14  6.28  33.52         23.0
1  109.97  1.04  27.01         20.7

1. 다중 선형 회귀 모델을 구축하고, 독립변수 o3의 회귀계수를 구하시오.
- 독립변수: solar(태양 에너지), wind(바람의 세기), o3(오존 농도)
- 종속변수: temperature(온도)


In [ ]:
from statsmodels.formula.api import ols
formula = "temperature ~ solar + wind + o3"
model = ols(formula, data=df).fit()
print(model.summary())
# 0.0749
model.params['o3']

                            OLS Regression Results                            
Dep. Variable:            temperature   R-squared:                       0.044
Model:                            OLS   Adj. R-squared:                  0.014
Method:                 Least Squares   F-statistic:                     1.464
Date:                Sat, 21 Sep 2024   Prob (F-statistic):              0.229
Time:                        05:45:45   Log-Likelihood:                -195.45
No. Observations:                 100   AIC:                             398.9
Df Residuals:                      96   BIC:                             409.3
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     19.0507      1.994      9.555      0.0

0.0749385437813658

2. 데이터에서 'solar'와 'o3' 값을 고정한 상태에서, 'wind'의 세기가 증가함에 따라 'temperature'가 감소하는지를 검증하기 위해 다중 선형 회귀 분석을 수행하고, 이 때'wind'의 회귀 계수에 대한 p-value 값을 구하시오. (유의수준: 0.05)

In [ ]:
# 이거 활용해서 기업 규모여부와 재계약의 상관관계등등 확인하면 좋을듯

In [ ]:
0.78
model.pvalues['wind']

0.7797177202071661

3. solar:100, wind:5, o3:30일 때 예측값과 그에 대한 95% 신뢰구간을 구하시오.

In [ ]:
new_data = pd.DataFrame({
    'solar':[100],
    'wind':[5],
    'o3':[30]
})
pred = model.get_prediction(new_data)
print(pred.summary_frame(alpha=0.05))

       mean   mean_se  mean_ci_lower  mean_ci_upper  obs_ci_lower  \
0  21.56163  0.175263      21.213737      21.909524     18.082985   

   obs_ci_upper  
0     25.040276  


###결과값을 하나씩 해석해드리겠습니다:

> 인용구 추가


1. mean (평균값): 21.56
solar: 100, wind: 5, o3: 30 조건에서 모델이 예측한 종속변수의 값이 21.56입니다.
종속 변수 = 온도('temperature')

2. mean_se (평균의 표준오차): 0.18
예측값의 표준오차가 0.18로 매우 작습니다.
이는 예측의 정밀도가 높다는 것을 의미합니다.

3. mean_ci_lower와 mean_ci_upper (평균의 신뢰구간):
하한값: 21.21
상한값: 21.91
95% 신뢰수준에서 실제 평균값이 21.21에서 21.91 사이에 있을 것으로 예상됩니다.
신뢰구간의 폭이 약 0.7로 매우 좁아, 예측의 정확도가 높음을 시사합니다.

4. obs_ci_lower와 obs_ci_upper (관측값의 신뢰구간):
하한값: 18.08
상한값: 25.04
새로운 관측값이 95% 확률로 18.08에서 25.04 사이에 있을 것으로 예상됩니다.

5.종합적 해석:

모델은 매우 정밀한 예측(21.56 ± 0.18)을 제공하고 있습니다.
평균에 대한 신뢰구간이 매우 좁아(21.21 ~
21.91) 예측의 신뢰성이 높습니다.
새로운 관측값의 예상 범위(18.08~25.04)는 평균의 신뢰구간보다 넓은데, 이는 개별 관측치의 자연스러운 변동성을 반영합니다.

In [ ]:
# 예측값 21.56163
# 예측값에 대한 95% 신뢰구간 21.213737      21.909524